workers/social/lib/social/models/computeproviders/stacktemplate.coffee

```diff
@@ -322,7 +339,76 @@ module.exports = class JStackTemplate extends Module
         callback err, this
 
 
-  generateStack: permit
+  shareWithTeam: permit
+
+    advanced: [
+      { permission: 'update own stack template', validateWith: Validators.own }
+      { permission: 'update stack template' }
+    ]
+
+    success: revive
+
+      shouldReviveClient   : yes
+      shouldReviveProvider : no
+      shouldFetchGroupLimit : yes
+
+    , (client, callback) ->
+
+      { group }    = client.r
+      { delegate } = client.connection
+      { stackTemplates } = group
+
+      stackTemplates = makeArray stackTemplates
+      id = @getId().toHexString()
+
+      if stackTemplates.length
+        unless id in stackTemplates
+          stackTemplates.push id
+      else stackTemplates = [id]
+
+      console.log 'stackTemplates', stackTemplates
+      @setAccess client, 'group', (err) =>
+        group.modify client, { stackTemplates }, (err) =>
+          return callback err  if err
+
+          return callback err, this
+
+
+  makePrivate: permit
+
+    advanced: [
+      { permission: 'update own stack template', validateWith: Validators.own }
+      { permission: 'update stack template' }
+    ]
+
+    success: revive
+
+      shouldReviveClient   : yes
+      shouldReviveProvider : no
+      shouldFetchGroupLimit : yes
+
+    , (client, callback) ->
+
+      { group }    = client.r
+      { delegate } = client.connection
+      { stackTemplates } = group
+
+      stackTemplates = makeArray stackTemplates
+      id = @getId().toHexString()
+
+      if stackTemplates.length
+        index = stackTemplates.indexOf(id)
+        if index > -1
+          stackTemplates.splice(index, 1)
+
+      @setAccess client, 'private', (err) =>
+        group.modify client, { stackTemplates }, (err) =>
+          return callback err  if err
+
+          return callback err, this
+
+
+  makeTeamDefault: permit
```

Comment:
this is not a feature of `JStackTemplate` it's a feature of `JGroup` and it already has a functionality for this https://github.com/koding/koding/blob/master/workers/social/lib/social/models/group/index.coffee#L1108-L1136